homeassistant/components/dominos.py

```diff
@@ -0,0 +1,251 @@
+
+    def __init__(self, hass, config):
+        """Set up main service."""
+        from pizzapi import Address, Customer, Store
+        self.hass = hass
+        self.customer = Customer(
+            config[DOMAIN].get(ATTR_FIRST_NAME),
+            config[DOMAIN].get(ATTR_LAST_NAME),
+            config[DOMAIN].get(ATTR_EMAIL),
+            config[DOMAIN].get(ATTR_PHONE),
+            config[DOMAIN].get(ATTR_ADDRESS))
+        self.address = Address(
+            *self.customer.address.split(','),
+            country=config[DOMAIN].get(ATTR_COUNTRY))
+        self.country = config[DOMAIN].get(ATTR_COUNTRY)
+        self._closest_store = Store()
+        self._last_store_check = 0
+        self.update_closest_store()
+
+    def handle_order(self, call=None):
+        """Handle ordering pizza."""
+        entity_ids = call.data.get(ATTR_ORDER_ENTITY, None)
+
+        target_orders = [order for order in self.hass.data[DOMAIN]['entities']
+                         if order.entity_id in entity_ids]
+
+        for order in target_orders:
+            order.place()
+
+    def update_closest_store(self):
+        """Update the shared closest store (if open)."""
+        cur_time = time.time()
+        if self._last_store_check + MIN_TIME_BETWEEN_STORE_UPDATES < cur_time:
+            self._last_store_check = cur_time
+            try:
+                self._closest_store = self.address.closest_store()
+            except Exception:
+                self._closest_store = False
+
+    @property
+    def closest_store(self):
+        """Return the shared closest store (or False if all closed)."""
+        return self._closest_store
+
+    @asyncio.coroutine
+    def show_menu(self, hass):
+        """Dump the closest stores menu into the logs."""
+        if self._closest_store is False:
+            _LOGGER.warning('Cannot get menu. Store may be closed')
+            return
+
+        menu = self._closest_store.get_menu()
+        hass.data[DOMAIN]['products'] = []
+
+        for product in menu.products:
+            item = {}
+            if isinstance(product.menu_data['Variants'], list):
+                variants = ', '.join(product.menu_data['Variants'])
+            else:
+                variants = product.menu_data['Variants']
+            item['name'] = product.name
+            item['variants'] = variants
+            hass.data[DOMAIN]['products'].append(item)
+
+
+class DominosProductListView(http.HomeAssistantView):
+    """View to retrieve product list content."""
+
+    url = '/api/dominos'
+    name = "api:dominos"
+
+    @callback
+    def get(self, request):
+        """Retrieve if API is running."""
+        return self.json(request.app['hass'].data[DOMAIN]['products'])
+
+
+class DominosOrder(Entity):
+    """Represents a Dominos order entity."""
+
+    def __init__(self, order_info, dominos):
+        """Set up the entity."""
+        self._name = order_info['name']
+        self.entity_id = async_generate_entity_id(
```

Comment:
There is no need to generate your own entity id based on name. The EntityComponent will do this for you.